contracts/minter-suite/Minters/MinterSEAV0.sol

```diff
@@ -0,0 +1,1186 @@
+     * `_bidTokenId` with bid amount and bidder address equal to
+     * `msg.value` and `msg.sender`, respectively.
+     * Intended to gracefully handle the case where a user is front-run by
+     * one or more transactions to settle and/or initialize a new auction,
+     * potentially still placing a bid on the auction for the token ID if the
+     * bid value is sufficiently higher than the current highest bid.
+     * Note that the use of `_targetTokenId` is to prevent the possibility of
+     * transactions that are stuck in the pending pool for long periods of time
+     * from unintentionally bidding on auctions for future tokens.
+     * Note that calls to `settleAuction` and `createBid` are possible
+     * to be called in separate transactions, but this function is provided for
+     * convenience and executes both of those functions in a single
+     * transaction, while handling front-running as gracefully as possible.
+     * @param _settleTokenId Token ID to settle auction for.
+     * @dev this function is not non-reentrant, but the underlying calls are
+     * to non-reentrant functions.
+     */
+    function settleAuctionAndCreateBid(
+        uint256 _settleTokenId,
+        uint256 _bidTokenId
+    ) external payable {
+        // ensure tokens are in the same project
+        require(
+            _settleTokenId / ONE_MILLION == _bidTokenId / ONE_MILLION,
+            "Only tokens in same project"
+        );
+        // settle completed auction, if applicable
+        settleAuction(_settleTokenId);
+        // attempt to bid on next token
+        createBid_l34(_bidTokenId);
+    }
+
+    /**
+     * @notice Settles a completed auction for `_tokenId`, if it exists and is
+     * not yet settled.
+     * Returns early (does not modify state) if
+     *   - there is no initialized auction for the project
+     *   - there is an auction that has already been settled for the project
+     *   - there is an auction for a different token ID on the project
+     *     (likely due to a front-run)
+     * This function reverts if the auction for `_tokenId` exists, but has not
+     * yet ended.
+     * @param _tokenId Token ID to settle auction for.
+     */
+    function settleAuction(uint256 _tokenId) public nonReentrant {
+        uint256 _projectId = _tokenId / ONE_MILLION;
+        ProjectConfig storage _projectConfig = projectConfig[_projectId];
+        Auction storage _auction = _projectConfig.activeAuction;
+        // CHECKS
+        // @dev this check is not strictly necessary, but is included for
+        // clear error messaging
+        require(_auction.initialized, "Auction not initialized");
+        if (_auction.settled || (_auction.tokenId != _tokenId)) {
+            // auction already settled or is for a different token ID, so
+            // return early and do not modify state
+            return;
+        }
+        require(block.timestamp > _auction.endTime, "Auction not yet ended");
```

Comment:
Nit: Would add a comment here clarifying that the order of these "require"-->"check for return early cases"-->"require" checks is important – as someone could accidentally move the ordering of these to consolidate requires and then return-early-checks and it would modify the intended behavior here.

Reply:
good call, updated with dev comment in commit [f322346](https://github.com/ArtBlocks/artblocks-contracts/pull/504/commits/f3223464d420c5224ae4726e28074b619cf38f84)